flexget/plugins/sites/rutracker.py

```diff
@@ -87,15 +84,15 @@ def update_base_url():
 
     def try_authenticate(self, payload):
         for _ in range(5):
-            s = RSession()
-            s.post('{}/forum/login.php'.format(self.base_url), data=payload)
-            if s.cookies and len(s.cookies) > 0:
-                return s.cookies
+            self.requests.post('{}/forum/login.php'.format(self.base_url), data=payload)
+            if self.requests.cookies and len(self.requests.cookies) > 0:
+                return self.requests.cookies
             else:
                 sleep(3)
         raise PluginError('unable to obtain cookies from rutracker')
 
-    def __init__(self, login, password, cookies=None, db_session=None):
+    def __init__(self, task, login, password, cookies=None, db_session=None):
+        self.requests = task.requests
```

Comment:
If you don't use the task for anything other than grabbing its requests session, why not cut out the middle man and just pass it directly?

Reply:
Fixed